frontend/src/components/FilesPage/services/helpers.ts

```diff
@@ -0,0 +1,167 @@
+                            sizeKb: 3320,
+                            type: 'file',
+                        },
+                    ],
+                },
+            ],
+        },
+        {
+            name: 'favorites',
+            sizeKb: 0,
+            type: 'dir',
+            items: [],
+        },
+        {
+            name: 'music',
+            sizeKb: 0,
+            type: 'dir',
+            items: [],
+        },
+        {
+            name: 'css',
+            sizeKb: 0,
+            type: 'dir',
+            items: [],
+        },
+        {
+            name: 'db',
+            sizeKb: 12476236523,
+            type: 'file',
+        },
+
+        {
+            name: 'README.md',
+            sizeKb: 0,
+            type: 'file',
+        },
+    ],
+};
+
+const fetchFolder = async (directoryParts: string[]) => {
+    let currentFolder: Folder = root;
+
+    for (const name of directoryParts) {
+        const foundItem = currentFolder.items.find(
+            (item) => item.name === name
+        );
+
+        // no item found
+        if (!foundItem) return null;
+
+        // found file
+        if (!('items' in foundItem)) return null;
+
+        currentFolder = foundItem;
+    }
+    return currentFolder;
+};
+
+export const useCurrentDirectory = () => {
+    const directory = useParams<'*'>()['*'] || '';
+
+    const directoryParts = useMemo(
+        () => directory.split('/').filter(Boolean),
+        [directory]
+    );
+
+    const [data, setData] = useState<Folder | null>(null);
+    const [isLoading, setIsLoading] = useState(true);
+
+    useEffect(() => {
+        setIsLoading(true);
+        fetchFolder(directoryParts).then((folder) => {
+            setData(folder);
+            setIsLoading(false);
+        });
+    }, [directoryParts]);
+
+    return {
+        isLoading,
+        directory: {
+            data,
+            directoryParts,
+        },
+    };
+};
+
+export const getBreadcumbs = (paths: string[]) => {
+    const breadcrumbs = [
+        {
+            label: 'My folder',
+            to: generateFolderPath(),
+        },
+    ];
+
+    for (const [index, label] of paths.entries())
+        breadcrumbs.push({
+            label,
+            to: generateFolderPath(paths.slice(0, index + 1)),
+        });
+
+    return breadcrumbs;
+};
+
+export const by =
+    <T extends Record<string, unknown>>(property: keyof T) =>
+    (left: T, right: T) => {
+        if (left[property] < right[property]) return -1;
+        if (left[property] > right[property]) return 1;
+        return 0;
+    };
+
+export const formatSize = (sizeKb: number) => {
+    const SCALE = 1000;
+    const sizeBytes = sizeKb / SCALE;
+    const i =
+        sizeBytes === 0 ? 0 : Math.floor(Math.log(sizeBytes) / Math.log(SCALE));
+    const unit = ['B', 'kB', 'MB', 'GB', 'TB'][i];
+    const value = Number.parseFloat(
+        (sizeBytes / Math.pow(SCALE, i)).toFixed(2)
+    );
+
+    return `${value} ${unit}`;
```

Comment:
I am sometimes seeing `unit` evaluate to `undefined`.

![image](https://user-images.githubusercontent.com/7527103/159356922-f57cf34c-9fba-4d51-921f-1f9ba636f696.png)

Reply:
Fixed in e7dfeb25b9777a0304a72833eb967364c263c90d. Test added in e7dfeb25b9777a0304a72833eb967364c263c90d